### ISSUE TITLE
Test `.snap` release

### ISSUE THREAD

**Issue by strogonoff** [paneron/paneron#79]
We had potential issues where a user couldn’t install `.snap`, could we check whether it works on Linux? I’m guessing Ubuntu is the most popular distribution.

Latest 2.x release can be downloaded from [Releases](https://github.com/paneron/paneron/releases), draft or published release is OK.

**Comment by ronaldtse** [comment]
I believe @CAMOBAP is the most familiar with the Snap system. Could you please help see how we can fix up this package so it is useable? I’m pretty sure we need to enable it to access the users home directory.

**Comment by strogonoff** [comment]
Yes, I talked to him and assigned after that. Glad to get your confirmation. @CAMOBAP also suggested an automated smoke-test of release on CI (not launching full GUI but via CLI means), if it’s relatively low effort we could add it in our GHA. But I have doubts it will cover these types of errors.

**Comment by CAMOBAP** [comment]
As far as I see we distribute snap only as a file in [`Releases`](https://github.com/paneron/paneron/releases) and don't publish it in snap's store like we do for metanrnorma.

Unfortunatelly due to known issue https://askubuntu.com/questions/822765/snap-install-failure-error-cannot-find-signatures-with-metadata-for-snap, snap cannot be installed without `--dangerous` flag

So to install snap from file it should be installed with command:

```
sudo snap install ./paneron-desktop-2.0.7.snap --dangerous
```

once installed rum `paneron` command from terminal, or search for `Paneron` in apps menu

2.0.7 version is works correctly I have successfully opened it an imported `mn-samples-iso`

So if we don't want to publish the snap soon, make sense to add this info to README

--

Regarding the minimal smoke test, unfortunately, electron doesn't provide any CLI out of the box, it runs GUI by default.

I propose to add some check for CLI args in https://github.com/paneron/paneron/blob/main/packages/core/src/main/index.ts

so the `paneron` app can handle `paneron --version`: print version and exit

@ronaldtse @strogonoff how do you think?

**Comment by strogonoff** [comment]
Thank you!

> So to install snap from file it should be installed with command:

Great to know the snap works, we’ll add these installation instructions before closing this issue.

@CAMOBAP is there a way to avoid having to pass `--dangerous`?

> I propose to add some check for CLI args in https://github.com/paneron/paneron/blob/main/packages/core/src/main/index.ts

That may work, but let me see if we can do a flag like `--check` to make sure the app has write access etc., to cover more potential installation issues.

**Comment by strogonoff** [comment]
@CAMOBAP Note that if you look at app code look in `release` branch. Releases are built from that branch, while `main` branch contains an unfinished migration to monorepo (it doesn’t build on CI properly yet). This is mentioned at the beginning of README.

**Comment by CAMOBAP** [comment]
> @CAMOBAP is there a way to avoid having to pass --dangerous?

The only is to publish the snap to https://snapcraft.io/ It will help to resolve this issue 

> That may work, but let me see if we can do a flag like --check to make sure the app has write access etc., to cover more potential installation issues.

Good point, I will prepare PR

**Comment by ronaldtse** [comment]
> The only is to publish the snap to https://snapcraft.io/

Yes we should do this. Thanks!

**Comment by CAMOBAP** [comment]
@ronaldtse I need your help with registering `paneron` snap. https://snapcraft.io/register-snap

I can do it on my own but we will need to "transfer" ownership then with post on forums, and we may waste time on this

We can register it for the existing `metanorma` organisation/user or create a new one dedicated for paneron

Also I will need tour help with add snapcraft's token to secrets https://snapcraft.io/docs/snapcraft-authentication

**Comment by strogonoff** [comment]
I wonder if there’s something to be done on this? If we are doing basic testing that sounds like what we planned, then publishing to Snapcraft would be a different issue…

**Comment by ronaldtse** [comment]
@strogonoff @CAMOBAP I've added the `paneron` snap on Snapcraft.

**Comment by CAMOBAP** [comment]
> I wonder if there’s something to be done on this? If we are doing basic testing that sounds like what we planned, then publishing to Snapcraft would be a different issue…

Agree

@ronaldtse could you please confirm that `secrets.SNAPCRAFT_TOKEN` or (`secrets.PANERON_SNAPCRAFT_TOKEN`) created

**Comment by CAMOBAP** [comment]
Ready for review https://github.com/paneron/paneron/pull/115
